src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1202,6 +1209,95 @@ DatabaseShardImp::findAcquireIndex(
     return boost::none;
 }
 
+void
+DatabaseShardImp::scheduleFinalizeShard(std::uint32_t shardIndex)
+{
+    taskQueue_->addTask([this, shardIndex]
```

Comment:
I don't see why this logic should be queued as a task.

Reply:
Removed with your patch.